modules/client/src/client-common/schemas/common.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  InvalidAddressError,
+  InvalidBigintError,
+  InvalidBigNumberishError,
+  InvalidDataError,
+  InvalidDateError,
+  InvalidIpfsUriError,
+  InvalidPermissionError,
+} from "@aragon/sdk-common";
+import { isAddress } from "@ethersproject/address";
+import { mixed, object, string } from "yup";
+import { Permissions } from "../../interfaces";
+import { isBigNumberish } from "@ethersproject/bignumber/lib/bignumber";
+import { isIpfsUri } from "../utils";
+
+export const addressSchema = string().notRequired().test(
+  "isAddress",
+  new InvalidAddressError().message,
+  (value) => value ? isAddress(value) : true,
+);
+
+export const ipfsUriSchema = string().notRequired().test(
+  "isIpfsUri",
+  new InvalidIpfsUriError().message,
+  (value) => value ? isIpfsUri(value) : true,
+);
+
+// address list or ens
+export const addressOrEnsSchema = string().test(
+  "isAddressOrEns",
+  new InvalidAddressError().message,
+  // check ens
+  (value) => value ? isAddress(value) : true,
+);
+// permissions
+export const permissionSchema = string().test(
+  "isPermission",
+  new InvalidPermissionError().message,
+  (value) => value ? Object.keys(Permissions).includes(value) : true,
+);
+
+// uint8array
+export const uint8ArraySchema = mixed().test(
+  "isUint8Array",
+  new InvalidDataError().message,
+  (value) => value ? value instanceof Uint8Array : true,
+);
+
+// uint8array
```

Comment:
Yes I briefly commented on this on my other comment, until we don't have a final `proposalId` format I used `BigNumber` because is what's on the contracts, so you can receive `5` `"5"` or `"0x5"` to refer to a `proposalId`. Once we decide and deploy the new format, this will be changed accordingly

Reply:
In general, the SDK should only have one correct way of receiving and returning values, and for big numbers this should be `bigint` and not `BigNumber`.

If we want to accept hex strings, then let's go all in and validate it